bayesian/utility/db_gateway.py

```diff
@@ -45,9 +45,19 @@ class GraphAnalyses:
             .as('version').in('has_version').dedup().as('package').select('version')
             .coalesce(out('has_snyk_cve').as('cve').select('package','version','cve')
             .by(valueMap()),select('package','version').by(valueMap()));
-            """
+            """,
     }
 
+    def __init__(self):
+        """Initialise GraphAnalyses."""
+        self.ca_batch_query = """
+            epv = [];packages.each {g.V().has('pecosystem', ecosystem).has('pname', it.name)
+            .has('version', it.version).as('version', 'cve').select('version').in('has_version')
+            .dedup().as('package').select('package', 'version', 'cve')
```

Comment:
For CA use case you may not need to traverse to Package node at all, Version and CVE node has all information.

Reply:
latest_non_cve_version, latest_version -> both reside at the package level. Not at the version level. I dont see how removing the pkg node would work here

Reply:
Agreed., my bad. :)